Src/Support/Google.Apis.Auth/OAuth2/BasicImpersonatedCredential.cs

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Auth.OAuth2.Requests;
+using Google.Apis.Http;
+using Google.Apis.Util;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Basic impersonated credential which obtains impersonated access tokens.
+    /// </summary>
+    public class BasicImpersonatedCredential : ServiceCredential, IGoogleCredential
+    {
+        private static readonly string[] s_impersonationScopes = new string[] { GoogleAuthConsts.IamScope };
+
+        /// <summary>An initializer class for the impersonated credential. </summary>
+        new public class Initializer : ServiceCredential.Initializer
+        {
+            /// <summary>
+            /// Gets or sets for how long the delegated credential should be valid.
+            /// Defaults to 1 hour or 3600 seconds.
+            /// </summary>
+            public TimeSpan Lifetime { get; set; }
+
+            /// <summary>
+            /// Gets or sets the source credential used to acquire the impersonated credentials.
+            /// </summary>
+            internal ICredential SourceCredential
+            {
+                get => HttpClientInitializers.OfType<ICredential>().SingleOrDefault();
+                set
+                {
+                    var sourceCredential = SourceCredential;
+
+                    if (value != null)
+                    {
+                        HttpClientInitializers.Add(ValidateAndScopeForImpersonation(value));
+                    }
+
+                    if (sourceCredential != null)
+                    {
+                        HttpClientInitializers.Remove(sourceCredential);
```

Comment:
It *looks* like we're adding a scoped version, but removing the unscoped version. That doesn't feel right...
I think I've now understood why, and it's because the property acts a bit oddly - you set one value, but then you'd receive a different one when fetching it.